src-tauri/src/producer.rs

```diff
@@ -13,30 +13,55 @@ use crate::{
     syndication::fetch_feed_items,
 };
 
-pub fn create_new_items(db: &Connection, proxy: Option<&str>) -> Vec<ItemToCreate> {
+use crate::error::{Error, Result};
+
+pub fn create_new_items(db: &Connection, proxy: Option<&str>) -> Result<Vec<ItemToCreate>> {
     let pairs = get_links_to_check(db);
 
     let mut inserted = vec![];
+
+    let feed_ids_to_check: Vec<i32> = pairs
+        .iter()
+        .filter_map(|(id, _, fetch_old_items)| if !fetch_old_items { Some(*id) } else { None })
+        .collect();
+
+    let most_recent_items = if !feed_ids_to_check.is_empty() {
+        get_most_recent_items(db, &feed_ids_to_check).unwrap_or_default()
+    } else {
+        HashMap::new()
+    };
+
     for (feed, link, fetch_old_items) in pairs {
-        let mut items = fetch_feed_items(&link, proxy).unwrap();
-
-        if !fetch_old_items {
-            let most_recent_items = get_most_recent_items(db);
-            if let Some(most_recent) = most_recent_items.get(&feed) {
-                items.retain(|item| {
-                    item.published_at
-                        .map_or(false, |published_at| published_at > *most_recent)
-                });
-            } else {
-                items.truncate(1)
-            }
+        let items = match fetch_feed_items(&link, proxy) {
+            Ok(items) => items,
+            Err(err) => return Err(Error::FetchFeedItemsFailure(err.to_string())),
+        };
```

Comment:
```suggestion
        let items = fetch_feed_items(&link, proxy)?;
```

I think it would be better to use a shortcut. If we need to provide an error that occurred while fetching items to the front-end, it would make sense to handle the error on L34 of `worker.rs`.

Reply:
I was ignorant of the `fetch_feed_items` function. I've made that change. Thanks.